python/eups/Eups.py

```diff
@@ -1026,6 +1026,16 @@ def findProductFromVRO(self, name, version=None, versionExpr=None, eupsPathDirs=
                 if self.cmdName != "setup":
                     print >> sys.stderr, "%-15s %s" % (product.name, product.version)
 
+        # Hack around Product.getTable() slowness, by internally caching Product instances
+        # FIXME: Should be removed once the Product.getTable() bottleneck is resolved
+        try:
+            product = self._productCache[product]
```

Comment:
Good point. I don't know whether the outputs are stable; I thought this would be less presumptuous and likely to cause breakage.

Actually, the right thing to do would be to change Product.getTable() (or Table() itself) to globally cache loaded tables. That seems more complex, I'd like to leave it for a subsequent PR.